src/load/azext_load/tests/latest/test_load_test.py

```diff
@@ -0,0 +1,500 @@
+    JMESPathCheck,
+    ResourceGroupPreparer,
+    ScenarioTest,
+    create_random_name,
+)
+
+rg_params = {
+    "name_prefix": "clitest-load-",
+    "location": "eastus",
+    "key": "resource_group",
+    "parameter_name": "rg",
+    "random_name_length": 30,
+}
+load_params = {
+    "name_prefix": "clitest-load-",
+    "location": "eastus",
+    "key": "load_test_resource",
+    "parameter_name": "load",
+    "resource_group_key": "resource_group",
+    "random_name_length": 30,
+}
+
+
+class LoadTestScenario(ScenarioTest):
+    def __init__(self, *args, **kwargs):
+        super(LoadTestScenario, self).__init__(*args, **kwargs)
+        self.kwargs.update({"subscription_id": self.get_subscription_id()})
+
+    @ResourceGroupPreparer(**rg_params)
+    @LoadTestResourcePreparer(**load_params)
+    def test_load_test_create(self, rg, load):
+        self.kwargs.update(
+            {
+                "test_id": LoadTestConstants.CREATE_TEST_ID,
+                "load_test_config_file": LoadTestConstants.LOAD_TEST_CONFIG_FILE,
+            }
+        )
+        checks=[JMESPathCheck("testId", self.kwargs["test_id"]),
+                    JMESPathCheck("loadTestConfiguration.engineInstances", 1),
+                    JMESPathCheck("environmentVariables.rps", '10')]
+        response = self.cmd(
+            "az load test create "
+            "--test-id {test_id} "
+            "--load-test-resource {load_test_resource} "
+            "--resource-group {resource_group} "
+            '--load-test-config-file "{load_test_config_file}" '
+            "--env rps=10 "
+            "--wait ",
+            checks=checks,
+        ).get_output_in_json()
+
+        tests = self.cmd(
+            "az load test list "
+            "--load-test-resource {load_test_resource} "
+            "--resource-group {resource_group}"
+        ).get_output_in_json()
+        assert self.kwargs["test_id"] in [test.get("testId") for test in tests]
+
+        self.cmd(
+            "az load test delete "
+            "--test-id {test_id} "
+            "--load-test-resource {load_test_resource} "
+            "--resource-group {resource_group} "
+            "--yes"
+        )
+
+    @ResourceGroupPreparer(**rg_params)
+    @LoadTestResourcePreparer(**load_params)
+    def test_load_test_list(self, rg, load):
+        self.kwargs.update(
+            {
+                "test_id": LoadTestConstants.LIST_TEST_ID,
+                "load_test_config_file": LoadTestConstants.LOAD_TEST_CONFIG_FILE,
+            }
+        )
+
+        create_test(self)
+
+        tests = self.cmd(
+            "az load test list "
+            "--load-test-resource {load_test_resource} "
+            "--resource-group {resource_group}"
+        ).get_output_in_json()
+
+        assert len(tests) > 0
+        assert "fake_test_id" not in [test["testId"] for test in tests]
+        assert self.kwargs["test_id"] in [test.get("testId") for test in tests]
+
+        delete_test(self)
```

Comment:
Recommend moving `delete_test(self)` in the `try... finally...`

Reply:
we removed all the delete_test as code cleaner take care of that. Resolving this